### DISCUSSION TITLE
Prometheus: Exemplars with linked traces shouldn't show exemplar modal, and should take user directly to the linked trace.

### DISCUSSION THREAD

**Discussion by gtk-grafana** [grafana/grafana#61714] [Observability]
Some users of Exemplars have reached out and asked if users can get from exemplar to associated trace by automatically opening the trace link on click of the exemplar (right now on click of the exemplar we "pin" the modal open, on hover the modal disappears when the user away from the modal.

The exemplar modals can be quite huge for some instances, and cumbersome to work with, so I see the benefit in pinning the exemplar, but I also understand in many use cases the details of the exemplar doesn't matter because it's already apparent by the context of the query, and the primary desired action is to jump to the trace.

Straw proposal:
If the user does not have traces properly configured with exemplars, then we should keep the behavior as it is (as there would be no link to the trace), and pin the exemplar on click. Otherwise, show the exemplar details on hover, but jump to the trace on click.

Questions:
Is this something that is beneficial to all users as a default UX? 
Or should we expose a configuration option to control this if we believe that pinning an exemplar is a valuable UX in certain use cases?

@leeoniya Is opening a new tab (or pane) from a click on an exemplar in canvas going to be a problem? I know google requires DOM events to drive navigation changes, not sure if the click on the canvas counts or not?

@diegoadams I'm not sure if we've done much research into the exemplar UX, but I'd be interested to hear your thoughts on this as well.

CC: @itsmylife @grafana/observability-metrics

**Comment by itsmylife** [comment]
We might want to give freedom to configure this. I know there will be a lot of configuration but I think some of the users might need to see all the fields. In the options section if the exemplars are enabled we show an extra configuration toggle about opening the trace links automatically. WDYT?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82811. Feel free to continue the discussion around this there. Thank you!
